api/api_public_transaction_pool.go

```diff
@@ -552,3 +552,46 @@ func resend(s *PublicTransactionPoolAPI, ctx context.Context, sendArgs NewTxArgs
 
 	return common.Hash{}, fmt.Errorf("Transaction %#x not found", matchTx.Hash())
 }
+
+// RecoverTransaction recovers the sender address from a signed raw transaction.
+// The signature is validated against the sender account's key configuration at the given block number.
+func (s *PublicTransactionPoolAPI) RecoverTransaction(ctx context.Context, encodedTx hexutil.Bytes, blockNumber rpc.BlockNumber) (common.Address, error) {
+	tx := new(types.Transaction)
+	if err := rlp.DecodeBytes(encodedTx, tx); err != nil {
+		return common.Address{}, err
+	}
+
+	signer := types.MakeSigner(s.b.ChainConfig(), s.b.CurrentBlock().Number())
+	state, _, err := s.b.StateAndHeaderByNumber(ctx, blockNumber)
+	if err != nil {
+		return common.Address{}, err
+	}
+	_, err = tx.ValidateSender(signer, state, blockNumber.Uint64())
+	if err != nil {
+		return common.Address{}, err
+	}
+	return signer.Sender(tx)
+}
+
+// RecoverMessage validates that the message is signed by one of the keys in the given account.
+// Returns the recovered signer address, which may be different from the account address.
+func (s *PublicTransactionPoolAPI) RecoverMessage(
```

Comment:
@aidan-kwon I made it return `boolean` first. but I thought there is a case the `pubkeyToAddress` or `pubkey` return value. Given a multisig key that includes 3keys and 3 persons have each key, dapp want to check how many persons signed the message. Returning boolean couldn't resolve this case.
In other words, It is to divide the duplicate sign.

Reply:
@kjeom How about changing the method name to `RecoverAddr`?

Reply:
If it's not a legacy key, I'm afraid it's confusing because it's not a real address. That's why I also considered returning the pubkey format. How do you think about `RecoverAddr`? @nohkwak @JayChoi1736 @blukat29

Reply:
`RecoverAddr` is ambiguous in that both RecoverTransaction and RecoverMessages return an address.
`RecoverMessage` might be problematic because in some Klaytn code "message" refers to "transaction". 
`RecoverSignature` or `RecoverSignedMessage` in the first comment looks more accurate.

Reply:
The method recovers an address from a message and a signature. I think `recoverAddr`, `recoverPubkey`, `recoverSigner`, or `recoverAddrFromMessage` are grammatically better.

Reply:
After discussion, we have renamed each as recoverFromTransaction and recoverFromMessage.